torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+#include <c10/util/complex.h>
+#include <torch/csrc/jit/codegen/cuda/arith.h>
+#include <torch/csrc/jit/codegen/cuda/ops/normalization.h>
+#include <torch/csrc/jit/codegen/cuda/python_frontend/fusion_definition.h>
+
+namespace nvfuser {
+struct RecordFunctor {
+  RecordFunctor(std::vector<size_t> _args, std::vector<size_t> _outputs)
+      : args(std::move(_args)), outputs(std::move(_outputs)) {}
+  virtual void operator()(FusionDefinition& fd) = 0;
+
+  std::vector<size_t> args;
+  std::vector<size_t> outputs;
+};
+
+struct InputTensorRecord : RecordFunctor {
```

Comment:
I think we have discussed this in the draft as well. I think having one input node per input tensor makes it unclear the order of inputs. https://github.com/kevinstephano/pytorch/pull/5#discussion_r917069446

Reply:
I did not consider aggregating the input and output records because the Records have a 1-to-1 correspondence to lines in the `FusionDefinition`.  There will be a general problem that another sequence of Records could have the identical information but be defined in a different order which will cause a mismatch.  However, solving general graph matching is a hard problem.  Therefore, giving a best effort to match definitions that should be ordered the same from block to block in code seems good enough.  

This is really just the recording of `FusionDefinition` lines.  Upon exit of the `FusionDefinition` context manager, an equality check needs to happen.  That could include aggregating the input and output records for matching.  Was there something you hand in mind that you think requires aggregation?

Reply:
^^^ That's fair. I was mostly coming from the perspective of looking at a computational graph. But you are right that our records is just a faithful recording of what happens during fusion difinition, I think it's fair for us to keep this as-is.

Coming out of the fusion definition world, we already have inputs/outputs aggregated at codgen IR level, so we should be all good.

Thanks for bearing with my ignorance. :bow: